include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -866,14 +866,31 @@ class alignment_file_output_format<format_bam> : alignment_file_output_format<fo
                     {
                         if (!std::ranges::empty(id_source)) // otherwise default will remain (-1)
                         {
-                            auto id_it = header.ref_dict.find(id_source);
+                            auto id_it = header.ref_dict.end();
+
+                            if constexpr (std::ranges::ContiguousRange<decltype(id_source)>)
+                            {
+                                id_it = header.ref_dict.find(std::span{std::ranges::data(id_source),
+                                                                       std::ranges::size(id_source)});
+                            }
+                            else
+                            {
+                                using header_ref_id_type = std::remove_reference_t<decltype(header.ref_ids()[0])>;
+
+                                static_assert(ImplicitlyConvertibleTo<decltype(id_source), header_ref_id_type>,
+                                  "The ref_id type is not convertible to the reference id information stored in the "
+                                  "reference dictionary of the header object.");
```

Comment:
Can you move this with the other static asserts / constraints?

Reply:
then I need to add two static asserts for ref_id and mate_ref_id instead of one in this lambda that processes both?

Reply:
Well we want to move the static asserts to the file anyway, right?

Reply:
some of them.. there are some that are format specific. e.g. that BAM requires the header member to be present

Reply:
ah and this assert cannot go into the file wither because how do I know on file level if the ref?id might be ignored? Because if it is ignored this check should not be performed!

Reply:
> some of them.. there are some that are format specific. e.g. that BAM requires the header member to be present

I thought we already discussed this and came to the conclusion that all the static_asserts are file by definition, because dynamic dispatch will lead to all formats being instantiated anyway.

Reply:
> I thought we already discussed this and came to the conclusion that all the static_asserts are file by definition, because dynamic dispatch will lead to all formats being instantiated anyway.

Yes I thought so but although they are instantiated it depends on how the read() function is called and thos requirements are different.. I noticed when I moved around some of the asserts in a temporary branch

Reply:
Hm, I am not quite sure I get it. In any case, please create a card for this.

Reply:
If we have a card for this, you can mark this as resolved.